MaxText/layers/linears.py

```diff
@@ -463,14 +508,22 @@ def gmm(inputs, kernel, group_sizes):
         output = output[: hs_shape[0]]
       return output
 
+    def get_all_to_all_params(all_shards_group_sizes, local_expert_size, num_expert_shards):
```

Comment:
Will you need transform_array function and TransformStrategy for other use cases except this function?

If they are only needed for this function, I suggest doing everything directly here with comments?

Reply:
Sounds good! So far, it only used in this case. Let me wrap those into this function.